Migration: Android v5 (#3528)
getsentry/sentry-docs: src/platforms/android/migration.mdx

## Migrating from `io.sentry:sentry` `4.3.0` to `io.sentry:sentry` `5.0.0`

You may remove `android:extractNativeLibs="true"` meta-data in the `AndroidManifest` file or `android.bundle.enableUncompressedNativeLibs=false` in the `gradle.properties` file if you're using the [Android Native Development Kit](/platforms/android/using-ndk/). Sentry can now symbolicate events with the default value of [extractNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#extractNativeLibs) and [android.bundle.enableUncompressedNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#behavior-changes).

`Sentry#startTransaction` by default does not bind created transaction to the scope. To start transaction with binding to the scope, use one of the new overloaded `startTransaction` methods taking `bindToScope` parameter and set it to `true`. Bound transaction can be retrieved with `Sentry#getSpan`.

A random generated `installationId` replaces `Settings.Secure.ANDROID_ID`, which has been removed. This may affect the number of unique users displayed on the the Issues page and Alerts.

> If you always set a custom user using `Sentry.setUser(customUser)`, the behavior has changed. While you don't have to do make any update, if you want to maintain the old behavior, use the following code snippet:

Author: ```suggestion
If you always set a custom user using `Sentry.setUser(customUser)`, the behavior has changed. While you don't have to make any update, if you want to maintain the old behavior, use the following code snippet:
```

## Migrating from `io.sentry:sentry` `4.3.0` to `io.sentry:sentry` `5.0.0`

If you're using the [Android Native Development Kit](/platforms/android/using-ndk/) support and you've set the `android:extractNativeLibs="true"` meta-data in the `AndroidManifest` file or `android.bundle.enableUncompressedNativeLibs=false` in the `gradle.properties` file, you don't have to anymore, you may remove them.

> Now Sentry is able to symbolicate events with the default value of [extractNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#extractNativeLibs) and [android.bundle.enableUncompressedNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#behavior-changes)

Author: @Swatinem review this part please, I guess we don't need to explain the reason, or do we?

> If you're using the [Android Native Development Kit](/platforms/android/using-ndk/) support and you've set the `android:extractNativeLibs="true"` meta-data in the `AndroidManifest` file or `android.bundle.enableUncompressedNativeLibs=false` in the `gradle.properties` file, you don't have to anymore, you may remove them.
> Now Sentry is able to symbolicate events with the default value of [extractNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#extractNativeLibs) and [android.bundle.enableUncompressedNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#behavior-changes)

Reviewer: ```suggestion
You may remove `android:extractNativeLibs="true"` meta-data in the `AndroidManifest` file or `android.bundle.enableUncompressedNativeLibs=false` in the `gradle.properties` file if you're using the [Android Native Development Kit](/platforms/android/using-ndk/). Sentry can now symbolicate events with the default value of [extractNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#extractNativeLibs) and [android.bundle.enableUncompressedNativeLibs](https://developer.android.com/studio/releases/gradle-plugin#behavior-changes).
```

`Sentry#startTransaction` by default does not bind created transaction to the scope. To start transaction with binding to the scope, use one of the new overloaded `startTransaction` methods taking `bindToScope` parameter and set it to `true`. Bound transaction can be retrieved with `Sentry#getSpan`.

> `Settings.Secure.ANDROID_ID` has been removed in favor of a random generated `installationId` and it may affect the number of unique users in the the Issues page and Alerts.

Reviewer: ```suggestion
A random generated `installationId` replaces `Settings.Secure.ANDROID_ID`, which has been removed. This may affect the number of unique users displayed on the the Issues page and Alerts.
```

> If you always set a custom user using `Sentry.setUser(customUser)`, you don't have to do anything, but in case you want to keep the old behaviour, you may copy the following code snippet.

Reviewer: ```suggestion
If you always set a custom user using `Sentry.setUser(customUser)`, the behavior has changed. While you don't have to do make any update, if you want to maintain the old behavior, use the following code snippet:
```

```java
User user = new User();
user.setId(Settings.Secure.ANDROID_ID);

Sentry.setUser(user);
```

> `SentryOptions#setEnableSessionTracking(boolean)` is deprecated in favor of `SentryOptions#setEnableAutoSessionTracking(boolean)` and it'll be removed in the future.

Reviewer: ```suggestion
`SentryOptions#setEnableSessionTracking(boolean)` is deprecated in favor of `SentryOptions#setEnableAutoSessionTracking(boolean)`. It will be removed in future.
```

_New_:

```java
SentryAndroid.init(this, options -> {
    options.setEnableAutoSessionTracking(false);
});
```

_Old_:

```java
SentryAndroid.init(this, options -> {
    options.setEnableSessionTracking(false);
});
```

> `io.sentry.session-tracking.enable` AndroidManifest meta-data is deprecated in favor of `io.sentry.auto-session-tracking.enable` and it'll be removed in the future.

Reviewer: ```suggestion
`io.sentry.session-tracking.enable` AndroidManifest meta-data is deprecated in favor of `io.sentry.auto-session-tracking.enable`. It will be removed in future.
```
